tephra-core/src/main/java/org/apache/tephra/txprune/TransactionPruningRunnable.java

```diff
@@ -57,8 +60,13 @@ public void run() {
       Transaction tx = txManager.startShort();
       txManager.abort(tx);
 
+      if (tx.getInvalids().length == 0) {
+        LOG.info("Invalid list is empty, not running transaction pruning");
```

Comment:
Yes - I was planning on making that change later since that needs some analysis. I'll file a JIRA for that. Is that okay?